addon/components/custom-subsidy-form-fields/climate-subsidy-costs-table/table-row-custom-data.js

```diff
@@ -0,0 +1,407 @@
+  @tracked toRealiseUnitsErrors = [];
+  @tracked costPerUnitErrors = [];
+
+  get storeOptions() {
+    return this.args.storeOptions;
+  }
+
+  // Do we need this for custom data?
+  get businessRuleUri() {
+    return new rdflib.NamedNode(this.args.businessRuleUriStr);
+  }
+
+  get climateTableSubject() {
+    return this.args.climateTableSubject;
+  }
+
+  // TODO: this logic probably needs to be adjusted
+  get hasValues() {
+    const values = this.storeOptions.store.match(
+      null,
+      actionDescriptionPredicate,
+      this.businessRuleUri,
+      this.storeOptions.sourceGraph
+    );
+    return values.length;
+  }
+
+  constructor() {
+    super(...arguments);
+    scheduleOnce('actions', this, this.initializeTableRow);
+  }
+
+  initializeTableRow() {
+    if (this.hasValues) {
+      this.loadProvidedValue();
+      this.args.updateTotalRestitution(this.restitution);
+      this.args.onUpdateRow();
+    } else {
+      this.initializeDefault();
+    }
+  }
+
+  loadProvidedValue() {
+    const values = this.storeOptions.store.match(
+      null,
+      actionDescriptionPredicate,
+      this.businessRuleUri,
+      this.storeOptions.sourceGraph
+    );
+    if (values.length > 1) {
+      throw `Expected single value for ${this.businessRuleUri}`;
+    } else {
+      this.setComponentValues(values[0].subject);
+    }
+  }
+
+  setComponentValues(subject) {
+    this.tableEntryUri = subject;
+    this.description = this.storeOptions.store.match(
+      this.tableEntryUri,
+      customDescriptionPredicate,
+      null,
+      this.storeOptions.sourceGraph
+    )[0].object.value;
+    this.amount = this.storeOptions.store.match(
+      this.tableEntryUri,
+      amountPerActionPredicate,
+      null,
+      this.storeOptions.sourceGraph
+    )[0].object.value;
+    this.costPerUnit = this.storeOptions.store.match(
+      this.tableEntryUri,
+      costPerUnitPredicate,
+      null,
+      this.storeOptions.sourceGraph
+    )[0].object.value;
+    this.restitution = this.storeOptions.store.match(
+      this.tableEntryUri,
+      restitutionPredicate,
+      null,
+      this.storeOptions.sourceGraph
+    )[0].object.value; // TODO: this seems to always return a string?
+    this.toRealiseUnits = this.storeOptions.store.match(
+      this.tableEntryUri,
+      toRealiseUnitsPredicate,
+      null,
+      this.storeOptions.sourceGraph
+    )[0].object.value;
+  }
+
+  initializeDefault() {
+    // The uuid should probably come from the parent component?
+    const uuid = uuidv4();
+    const tableEntryUri = new rdflib.NamedNode(`${tableEntryBaseUri}/${uuid}`);
+
+    let triples = [
+      {
+        subject: tableEntryUri,
+        predicate: RDF('type'),
+        object: ClimateEntryType,
+        graph: this.storeOptions.sourceGraph,
+      },
+      {
+        subject: tableEntryUri,
+        predicate: MU('uuid'),
+        object: uuid,
+        graph: this.storeOptions.sourceGraph,
+      },
```

Comment:
As mentioned before, this is the reason why we would need to pass in the uuid from the parent component, but I'm not sure if it's really needed since we won't use mu-cl-resources to fetch this?